panel/models/vtk.ts

```diff
@@ -1,60 +1,90 @@
 import * as p from "core/properties"
+import {clone} from "core/util/object";
 import {HTMLBox, HTMLBoxView} from "models/layouts/html_box";
 import {div} from "core/dom";
 
 export class VTKPlotView extends HTMLBoxView {
   model: VTKPlot
   protected _vtk: any
-  protected _rendererEl: any
   protected _container: HTMLDivElement
-  protected _vtk_initialized: boolean
+  protected _rendererEl: any
+  protected _renderer: any
+  protected _camera: any
+  protected _setting: boolean = false
 
   initialize(): void {
     super.initialize()
-    const width = this.model.width ? "100%" : "300px"
-    const height = this.model.height ? "100%" : "300px"
     this._vtk = (window as any).vtk
     this._container = div({
       style: {
-        width,
-        height
+        width: "100%",
+        height: "100%"
       }
-    })
-    this._vtk_initialized = false
+    });
   }
 
-
-  after_layout() {
+  after_layout(): void {
     super.after_layout()
-    if (!this._vtk_initialized) {
-        this._rendererEl = this._vtk.Rendering.Misc.vtkFullScreenRenderWindow.newInstance({
-          rootContainer: this.el,
-          container: this._container
-        });
-        this._plot()
-        this._vtk_initialized = true
+    if (!this._rendererEl) {
+      this._rendererEl = this._vtk.Rendering.Misc.vtkFullScreenRenderWindow.newInstance({
+        rootContainer: this.el,
+        container: this._container
+      });
+      this._renderer = this._rendererEl.getRenderer()
+      this._camera = this._renderer.getActiveCamera()
+      this._plot()
+      this._camera.onModified(() => this._get_camera_state())
     }
   }
 
+  connect_signals(): void {
+    super.connect_signals()
+    this.connect(this.model.properties.data.change, () => this._plot())
+    this.connect(this.model.properties.camera.change, () => this._set_camera_state())
+  }
+
   render() {
     super.render()
-    this.el.appendChild(this._container)
+    if (!(this._container === this.el.childNodes[0]))
+      this.el.appendChild(this._container)
   }
 
-  connect_signals(): void{
-    this.connect(this.model.properties.vtkjs.change, this._plot)
+  _get_camera_state(): void {
+    if (!this._setting) {
+      this._setting = true;
+      const state = clone(this._camera.get());
+      delete state.classHierarchy;
+      delete state.vtkObject;
+      delete state.vtkCamera;
+      delete state.viewPlaneNormal;
+      this.model.camera = state;
+      this.model.properties.camera.change.emit();
```

Comment:
Is it really necessary to emit change explicitly?
As we set a new value to model.camera, I would excpect the change event would be already emit

Reply:
I think you're right, thanks!